src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,171 @@
 package core.basesyntax;
-
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static class Node<T> {
+        T item;
+        Node<T> next;
+        Node<T> prev;
+
+        public Node(T item, Node<T> next, Node<T> prev) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            if (current == null) {
+                throw new NullPointerException();
+            }
+            current = current.next;
+        }
+        return current;
+    }
+
+
+    private int findIndexByValue(T value) {
+        Node<T> current = head;
+        int index = 0;
+        while (current != null) {
+            if (Objects.equals(value, current.item)) {
+                return index;
+            }
+            current = current.next;
+            index++;
+        }
+        return index;
+    }
+
+    private void unlink(Node<T> node) {
+        if (node.prev == null) {
+            head = node.next;
+        } else {
+            node.prev.next = node.next;
+            if (node.next != null) {
+                node.next.prev = node.prev;
+            } else {
+                tail = node.prev;
+            }
+        }
+        size--;
+    }
```

Comment:
In the `unlink` method, when `node.prev` is not `null`, you should also check if `node.next` is not `null` before setting `node.prev.next`. This ensures that you're not trying to access a property on a `null` reference.